### ISSUE TITLE
Select crafting recipe based on input availability

### ISSUE THREAD

**Issue by TheRealWormbo** [Astavie/ThermalLogistics#19]
It would be nice to be able to define multiple combinations of input items for a crafter recipe slot with identical outputs, which can be prioritized in multiple ways.

Two examples:
- In Thermal Expansion, electrum can be created in the induction smelter using gold and silver in any combination of pulverized and ingot form. It might make sense to prefer pulverized input metals, as that would have a lower power consumption (unless they need to be pulverized first, but I guess that's an additional preference of "available" over "craftable").
- In Botania, mana powder can be created from a number of different items, like gunpowder, glowstone dust, sugar, etc., by dropping it into a mana pool. A reasonable strategy here could be to use the input item with the highest number of available items.

So basically there should be prioritization such that recipes with unavailable inputs are favored least, and among the remaining recipes, the user could pick a strategy like "most available input", "first in list", "nearest input", "round robin", etc.

**Comment by Krougal** [comment]
Fuzzy crafting. Yes, I've been begging for this as well.
Some of it isn't possible unless the device supports it also (for instance Sequential Fabricator can only do 1 recipe) but at least for most machine inputs or outputs this is viable.

Needs to be selectable Input & Output (in, out or both) side as well.
Fuzzy input - your examples
Fuzzy output - we want planks from the sawmill and we don't care what kind of planks. 
Both - I can't really think of a practical example right now, but let's say stained glass panes. We don't care what color we get out or maybe we just want a stockpile of all the colors. It is a different recipe whatever we put in (fabricator would need support for this, although I think there are some other mods crafting tables that work this way)...so make glass with whatever dye is handy.

Some of this may already be possible with NTB/oredict filters, I have not experimented enough yet (on my todo).

**Comment by Astavie** [comment]
@ldog I'm not sure what fuzzy crafting has to do with this issue? But to elaborate on what you said: fuzzy crafting can be done with the nbt/oredict filters, but right now the filters are applied to both input and output, not separately.

**Comment by Astavie** [comment]
Or it might only apply to the input, I'm actually not sure

**Comment by Astavie** [comment]
Just checked: nbt/oredict configure the fuzzyness of the input

**Comment by Krougal** [comment]
Thanks for checking, it's on my todo but I just haven't gotten to it. This is Krougal BTW.

Aren't all of these forms of fuzzy crafting? 
Yes, re-reading a few times and thinking it over, Wormbo is asking for a slightly different feature, which I'd like as well.
Although, once again I have not done my due diligence in understanding how the prioritization is being done currently. I imagine it is nearest-first, so this could be controlled easily enough by proper routing and use of vacuum/dense ducts.

**Comment by Krougal** [comment]
Here's something more in line with the op, and probably a bit extreme. 
Making destabilized redstone in magma crucible. 4 possible inputs, same output but different amounts. So maybe I don't really care how much I get, I just need some, and need it made out of whatever is on-hand.

That could get very messy very fast though, so I understand this isn't a simple request by any means.
